lib/datasource/deb/index.ts

```diff
@@ -0,0 +1,306 @@
+   */
+  override readonly customRegistrySupport = true;
+
+  /**
+   * The original apt source list file format is
+   * deb uri distribution [component1] [component2] [...]
+   * @see{https://wiki.debian.org/DebianRepository/Format}
+   *
+   * However, for Renovate, we require the registry URLs to be
+   * valid URLs which is why the parameters are encoded in the URL.
+   *
+   * The following query parameters are required:
+   * - components: comma separated list of components
+   * - suite: stable, oldstable or other alias for a release, either this or release must be given
+   * - release: buster, etc.
+   */
+  override readonly defaultRegistryUrls = [
+    'https://ftp.debian.org/debian?suite=stable&components=main,contrib,non-free',
+  ];
+
+  override readonly defaultConfig: DebLanguageConfig = {
+    deb: {
+      binaryArch: 'amd64',
+      downloadDirectory: './others/deb/download',
+      extractionDirectory: './others/deb/extracted',
+    },
+  };
+
+  override readonly caching = false; // TODO: how can this be used?
+
+  /**
+   * Here, we tell Renovate that this data source can respect multiple upstream repositories
+   */
+  override readonly registryStrategy = 'merge';
+
+  /**
+   * Not all Debian packages follow Semver, so it's wise to keep this loose but make sure to
+   * have enough tests in your application.
+   */
+  override readonly defaultVersioning = 'loose';
+
+  requiredPackageKeys = ['Package', 'Version', 'Homepage'];
+
+  async initCacheDir(cfg: DebLanguageConfig): Promise<void> {
+    await fs.ensureCacheDir(cfg.deb.downloadDirectory);
+    await fs.ensureCacheDir(cfg.deb.extractionDirectory);
+  }
+
+  async extract(
+    compressedFile: string,
+    compression: string,
+    outputFile: string
+  ): Promise<void> {
+    if (compression === 'gz') {
+      const source = fs.createReadStream(compressedFile);
+      const destination = fs.createWriteStream(outputFile);
+      await fs.pipeline(source, createUnzip(), destination);
+    } else {
+      throw 'Unknown compression standard, this is probably a programming error';
+    }
+  }
+
+  async downloadAndExtractPackage(
+    cfg: DebLanguageConfig,
+    basePackageUrl: string
+  ): Promise<string> {
+    // we hash the package URL and export the hex to make it file system friendly
+    // we use the hashed url as the filename for the local directories/files
+    const hash = createHash('sha256');
+    hash.update(basePackageUrl);
+    const hashedPackageUrl = hash.copy().digest('hex');
+    const downloadDirectory = await fs.ensureCacheDir(
+      cfg.deb.downloadDirectory
+    );
+
+    const extractionDirectory = await fs.ensureCacheDir(
+      cfg.deb.extractionDirectory
+    );
+    const extractedFile = extractionDirectory + '/' + hashedPackageUrl + '.txt';
+    let extractedFileExists = false;
+    try {
+      await fs.access(extractedFile, fs.constants.R_OK);
+      extractedFileExists = true;
+    } catch (e) {
+      //ignore
+    }
```

Comment:
Can be much simplier:
```suggestion
    let extractedFileExists = fs.pathExists(extractedFile);
```

Checkout `util/fs/proxies` 😉 
https://github.com/renovatebot/renovate/blob/c5d50f6da3502d9f2a8f805fb65292a98246a6bf/lib/util/fs/proxies.ts#L60-L63